src/test/kotlin/com/github/rushyverse/PvpboxServerTest.kt

```diff
@@ -0,0 +1,148 @@
+import com.github.rushyverse.pvpbox.listener.death.PlayerDeathListener
+import com.github.rushyverse.pvpbox.listener.item.PlayerDropItemListener
+import com.github.rushyverse.pvpbox.listener.item.PlayerInventoryClickListener
+import com.github.rushyverse.pvpbox.listener.item.PlayerItemClickListener
+import com.github.rushyverse.pvpbox.listener.item.PlayerSwapItemListener
+import com.github.rushyverse.utils.randomString
+import io.mockk.mockk
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.test.runTest
+import net.minestom.server.MinecraftServer
+import net.minestom.server.entity.Player
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.assertThrows
+import java.io.IOException
+import kotlin.test.*
+
+class PvpboxServerTest : AbstractTest() {
+
+    @AfterTest
+    override fun onAfter() {
+        super.onAfter()
+        MinecraftServer.stopCleanly()
+    }
+
+    @Nested
+    inner class CreateOrGetConfiguration {
+
+        @Test
+        fun `should create a configuration file if it doesn't exist`() = runTest {
+            assertThrows<IOException> {
+                PvpboxServer().start()
+            }
+            val configurationFile = fileOfTmpDirectory(IConfiguration.DEFAULT_CONFIG_FILE_NAME)
+            assertTrue { configurationFile.isFile }
+
+            val configuration = IConfiguration.readHoconConfigurationFile<PvpboxConfiguration>(configurationFile)
+            assertEquals(expectedDefaultConfiguration, configuration)
+        }
+
+        @Test
+        fun `should use the configuration file if exists`() = runTest {
+            val configurationFile = fileOfTmpDirectory(randomString())
+            assertTrue { configurationFile.createNewFile() }
+
+            val configuration = defaultConfigurationOnAvailablePort()
+            configurationToHoconFile(configuration, configurationFile)
+
+            val exception = assertThrows<FileSystemException> {
+                PvpboxServer(configurationFile.absolutePath).start()
+            }
+            assertEquals(configuration.server.world, exception.file.name)
+        }
+
+    }
+
+    @Nested
+    inner class UseConfiguration {
+
+        @Test
+         fun `should use configuration to turn on the server`()  {
+            val configuration = defaultConfigurationOnAvailablePort()
+            val configurationFile = fileOfTmpDirectory(randomString())
+            configurationToHoconFile(configuration, configurationFile)
+
+            copyWorldInTmpDirectory(configuration)
+
+            runBlocking { PvpboxServer(configurationFile.absolutePath).start() }
+
+            // If no exception is thrown, the world is loaded
+            assertTrue { MinecraftServer.isStarted() }
+
+            val server = MinecraftServer.getServer()
+            assertEquals(configuration.server.port, server.port)
+            assertEquals("0.0.0.0", server.address)
+        }
+    }
```

Comment:
Should use the test method to have a coroutine context
```suggestion
        @Test
         fun `should use configuration to turn on the server`() = runTest {
            val configuration = defaultConfigurationOnAvailablePort()
            val configurationFile = fileOfTmpDirectory(randomString())
            configurationToHoconFile(configuration, configurationFile)

            copyWorldInTmpDirectory(configuration)

            PvpboxServer(configurationFile.absolutePath).start()

            // If no exception is thrown, the world is loaded
            assertTrue { MinecraftServer.isStarted() }

            val server = MinecraftServer.getServer()
            assertEquals(configuration.server.port, server.port)
            assertEquals("0.0.0.0", server.address)
        }
    }
```

Reply:
https://github.com/Rushyverse/pvpbox/commit/02c8fac5a13d4982069395453db6f90084d3d6af